api/tests/Page/DownloadTest.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace SynchWeb;
+
+use PHPUnit\Framework\TestCase;
+use SynchWeb\Page\Download;
+use React\EventLoop\Factory;
+use React\ChildProcess\Process;
+use ZipArchive;
+
+
+final class DownloadTest extends TestCase
+{
+
+    private function countFilesInZipStream($zipFileStream)
+    {
+        $zip = new ZipArchive();
+        if ($zip->open($zipFileStream) !== true) {
+            return 0;
+        }
+
+        $numFiles = $zip->numFiles;
+        $zip->close();
+
+        return $numFiles;
+    }
+
+    private function getZipStreamFileContents($zipFileStream)
+    {
+        $zip = new ZipArchive();
+        if ($zip->open($zipFileStream) !== true) {
+            return [];
+        }
+
+        $fileContents = [];
+        for ($i = 0; $i < $zip->numFiles; $i++) {
+            $filename = $zip->getNameIndex($i);
+            $fileContents[$filename] = $zip->getFromIndex($i);
+        }
+
+        $zip->close();
+
+        return $fileContents;
+    }
+
+    /**
+     * @runInSeparateProcess
+     */
+    public function testDownload()
+    {
+
+        // Download the zip file to a memory stream
+        $loop = Factory::create();
+
+        // Call the class method in a separate process.
+        $process = new Process("php tests/Page/DownloadRun.php");
+
+        $process->start($loop);
+
+        $process->stdout->on('data', function ($result) {
+            if ($result !== true) {
+                switch ($result) {
+                    case ZipArchive::ER_NOZIP:
+                        $this->fail('The file is not a zip archive.');
+                        break;
+                    case ZipArchive::ER_INCONS:
+                        $this->fail('The zip archive is inconsistent.');
+                        break;
+                    case ZipArchive::ER_CRC:
+                        $this->fail('The file has a CRC error.');
+                        break;
+                    default:
+                        $this->fail("Failed to open the zip file. Zip file contents: " . $result);
```

Comment:
It fails here with a message:
Failed to open the zip file. Zip file contents: P�H�V�[W�       file1.txt
                                                                         ��,V��̜TC=